### ISSUE TITLE
Vertical center-align of single lines in double lined subtitles.

### ISSUE THREAD

**Issue by nnuel** [libass/libass#588]
I am trying to show my subtitles in the black box under the video. Here is a good example of it: https://github.com/mpv-player/mpv/issues/3008

![image](https://user-images.githubusercontent.com/4148154/152122669-c753d86c-7520-4cfc-b9d9-7b157fc2f99b.png)

On the issue it is said that this has to be changed within libass. Is that feasible to make it like that for when the blackbox is big enough to display two lines of subtitles. If the blackbox is high enough for one line, than the bottom-aligned option works best, and also when no black box present the default option works best. It is just for these wide-screen videos on a 16:9 screen that leave a big black box that the subtitles pushed to the bottom look like it could be improved by better alignment.

**Comment by astiob** [comment]
This sounds like a feature request for changing how subtitles are displayed in margins (screen space outside the video).

On the one hand, I think this would be feasible, but not quite trivial, to implement: when positioning top- or bottom-aligned events, we could check for margins and take a separate branch that computes the anchor point’s position such that the whole event fits in the margin minus the event’s `MarginV` and is centred within it if possible, or if not, falls back to regular positioning at the edge of the screen.

On the other hand, this raises a whole host of further questions/problems:

* Does everyone want this?
* If a second subtitle comes on screen while one is already displayed (or if multiple subtitle events appear at the same time), this will effectively waste space at the edge of screen and push the second subtitle too far into the video’s direction.
* If the margin is very spacious, putting subtitles in its centre may perhaps look weird.
* This will look weird when the video is zoomed in in mpv and begins to occupy the previously-black area.
* Should they be centred within the full margin or `(margin - MarginV)`? I’m assuming we don’t want to let subtitles go off-screen and we still want to honour `MarginV`, which is the minimum distance from the subtitles to the edge of the screen (defined in the subtitle file or the user’s style overrides).

If we’re talking about improving text positioning in margins, I also want to plug my madVR screenshots in https://github.com/libass/libass/pull/384#issuecomment-605161837, which I still think does this even better: instead of placing the subtitles at the bottom of the screen or in the middle of the black bar, it places them just below the video, so the eye doesn’t have to travel as far. It does have some of the same issues I listed above, though.

**Comment by waot** [comment]
Yes, having it anchored to the bottom of the video and therefore at the top of the black box would be best for really big black boxes. Indeed the eye-travel is what you want to minimize and that's rather tricky when they are anchored or calculated from the bottom of the screen by default. But also having the center option would be good. It's like how `left | center | right` options are offered for the horizontal alignment, but then there could be a `top | center | bottom`, for the alignment in the margins/blackbox, whenever they are large enough to fit. I can see wherever I can help out with testing, but as of right now my programming experience in libass is non-existent.

**Comment by dyphire** [comment]
> instead of placing the subtitles at the bottom of the screen or in the middle of the black bar, it places them just below the video, so the eye doesn’t have to travel as far.

That sounds good.
